src/tree.service.ts

```diff
@@ -98,4 +107,18 @@ export class TreeService {
   public hasController(id: string | number): boolean {
     return this.controllers.has(id);
   }
+
+  private shouldFireLoadNextLevel(tree: Tree): boolean {
+
+    const result = tree.node.hasChildren &&
```

Comment:
@TMaster seems like not all the branches are covered in this condition - by adding more tests (until yellow mark turned into green one), you will fix this. Also have a look at [utility functions](https://github.com/valor-software/ng2-tree/blob/master/src/utils/fn.utils.ts)  that you can use here. For example instead of `tree.children === []`, you can use `isEmpty(tree.children)`

Reply:
Hi @rychkog, what do you mean by **until yellow mark turned into green one**, what marks?

Thanks

Reply:
@TMaster Ah, sorry for that :)) I have a codecov plugin that shows me covered lines in the PR :) Try it out - it might be very helpful :)
https://chrome.google.com/webstore/detail/codecov-extension/keefkhehidemnokodkdkejapdgfjmijf

Reply:
![image](https://user-images.githubusercontent.com/1942440/31536490-77e730b6-b007-11e7-8f51-aa0c332656dd.png)